lib/net.js

```diff
@@ -220,20 +224,24 @@ function Socket(options) {
     this._handle = options.handle; // private
     this[async_id_symbol] = getNewAsyncId(this._handle);
   } else if (options.fd !== undefined) {
-    this._handle = createHandle(options.fd, false);
-    this._handle.open(options.fd);
+    const fd = options.fd;
+    this._handle = createHandle(fd, false);
+    this._handle.open(fd);
     this[async_id_symbol] = this._handle.getAsyncId();
     // options.fd can be string (since it is user-defined),
     // so changing this to === would be semver-major
     // See: https://github.com/nodejs/node/pull/11513
     // eslint-disable-next-line eqeqeq
-    if ((options.fd == 1 || options.fd == 2) &&
+    if ((fd == 1 || fd == 2) &&
         (this._handle instanceof Pipe) &&
         process.platform === 'win32') {
       // Make stdout and stderr blocking on Windows
       var err = this._handle.setBlocking(true);
       if (err)
         throw errnoException(err, 'setBlocking');
+
+      this._writev = null;
+      this._write = makeSyncWrite(fd);
```

Comment:
Can we assign `fd` to a symbol property here and do the if logic in `Socket.prototype._write` to avoid the additional closure?

Reply:
@joyeecheung I have thought about making `fd` a property for all `net.Socket`s, but I’m not sure whether that’s a good idea or not.

We could add a symbol, but for now I’d probably leave the code as it is, especially given that the worst case is that this might create 2 closures for the entire process…